File: solutions/free-pascal/sum-of-multiples/1/SumOfMultiples.pas
unit SumOfMultiples;

{$mode ObjFPC}{$H+}

interface

function sum(const factors : Array Of Integer; const limit : integer) : integer;

implementation

uses SysUtils;

function sum(const factors : Array Of Integer; const limit : integer) : integer;
var n,f : integer;
begin
result := 0;
for n:=1 to limit-1 do
begin
  for f in factors do
  begin
    if f=0 then
      continue;
    if n mod f = 0 then
    begin
      result += n;
      break;
    end;
  end;
end;

end;

end.

File: solutions/cpp/sum-of-multiples/7/sum_of_multiples.cpp
#include "sum_of_multiples.h"

#include <initializer_list>

namespace sum_of_multiples {

    int to(std::initializer_list<int> factors, int limit){
        int sum{0};
        for(int m=1; m<limit; m++){
            for(int f : factors){
                    if (f!=0 && m % f==0) {
                        sum +=m;
                        break;
                    }
            }
        }
        return sum;
    }
    

}  // namespace sum_of_multiples
